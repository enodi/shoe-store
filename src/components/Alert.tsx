import {
  Alert,
  AlertIcon,
  AlertTitle,
  AlertDescription,
  CloseButton
} from "@chakra-ui/react"

interface Props {
  handleOnClick?: () => void;
  title: string;
  description?: string;
  status?: "success" | "warning" | "error" | "info";
  closeButton?: boolean;
}

const AlertComponent:React.FC<Props> = ({
  handleOnClick,
  title,
  description,
  status = "info",
  closeButton = false
}) => (
  <Alert status={status} sx={{mb: 10}}>
    <AlertIcon />
    <AlertTitle mr={2}>{title}</AlertTitle>
    <AlertDescription>{description}</AlertDescription>
    {closeButton && <CloseButton position="absolute" right="8px" top="8px" onClick={handleOnClick}/>}
  </Alert>
)

export default AlertComponent;
