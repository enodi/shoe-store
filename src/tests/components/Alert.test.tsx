import { shallow } from "enzyme";
import Alert from "../../components/Alert";

describe("Alert Component", () => {
  let wrapper: any, handleOnClick: () => void;;
  beforeEach(() => {
    handleOnClick = jest.fn();
    wrapper = shallow(<Alert
      handleOnClick={handleOnClick}
      title="Network Error!"
      description="A network error occurred. Please try reconnecting again"
      status="error"
      closeButton={true}
    />);
  });

  it("should render Alert Error component correctly", () => {
      expect(wrapper).toMatchSnapshot();
  });

  test("should call handleOnClick on button click", () => {
    wrapper.find("CloseButton").simulate("click");
    expect(handleOnClick).toHaveBeenCalled();
});
});