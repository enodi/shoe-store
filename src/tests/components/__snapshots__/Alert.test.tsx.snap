// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component should render Alert Error component correctly 1`] = `
<ForwardRef
  status="error"
  sx={
    Object {
      "mb": 10,
    }
  }
>
  <AlertIcon />
  <ForwardRef
    mr={2}
  >
    Network Error!
  </ForwardRef>
  <ForwardRef>
    A network error occurred. Please try reconnecting again
  </ForwardRef>
  <CloseButton
    onClick={[MockFunction]}
    position="absolute"
    right="8px"
    top="8px"
  />
</ForwardRef>
`;
